python/beeai_framework/agents/requirement/events.py

```diff
@@ -17,7 +18,15 @@ class RequirementAgentSuccessEvent(BaseModel):
     response: ChatModelOutput
 
 
+class RequirementAgentFinalAnswerEvent(BaseModel):
```

Comment:
What if we refactored these event models to introduce a base class for common fields?

```Python
class RequirementAgentEvent(BaseModel):
    state: RequirementAgentRunState


class RequirementAgentStartEvent(RequirementAgentEvent):
    request: RequirementAgentRequest


class RequirementAgentSuccessEvent(RequirementAgentEvent):
    response: ChatModelOutput


class RequirementAgentFinalAnswerEvent(RequirementAgentEvent):
    output_structured: BaseModel | Any
    output: str
    delta: str
```